contrib/libarchive-cmake/CMakeLists.txt

```diff
@@ -135,10 +135,18 @@ set(SRCS
 )
 
 add_library(_libarchive ${SRCS})
-target_include_directories(_libarchive PUBLIC 
-    ${CMAKE_CURRENT_SOURCE_DIR}
-    "${LIBRARY_DIR}/libarchive"
-)
+if (OS_SUNOS)
+	target_include_directories(_libarchive PUBLIC
+	    ${CMAKE_CURRENT_SOURCE_DIR}/illumos
+	    "${LIBRARY_DIR}/libarchive"
+	    "/usr/include/libxml2"
```

Comment:
This won't work for cross-compilation and for hermetic builds. Let's find where libxml2 is inside our source tree...

Reply:
Ah yes, I'll come back to this one, thanks.